### PR TITLE
CC | protocols: add support sealed_secret

### PR DESCRIPTION
**Pull request by Lu-Biao**
To call CDH ttrpc API, 'unseal_secret' for 'sealed_secret', add protocol file and generate ttrpc code.

Fixes: #7544

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Lu-Biao!

/cc @arronwy

**Comment by fidencio** [comment]
I'm adding the `do-not-merge` label just till we finish the merge from main to CCv0, which should be done Today.

**Review by GabyCT — APPROVED** [review]

**Comment by fidencio** [comment]
@Lu-Biao, our static checks are not happy:
```
INFO: cargo fmt...
cargo fmt -- --check
Diff in /home/runner/work/kata-containers/kata-containers/src/github.com/kata-containers/kata-containers/src/libs/protocols/build.rs at line 214:
         fs::rename("src/health_ttrpc.rs", "src/health_ttrpc_async.rs")?;
         fs::rename("src/image_ttrpc.rs", "src/image_ttrpc_async.rs")?;
         #[cfg(feature = "sealed-secret")]
-        fs::rename("src/sealed_secret_ttrpc.rs", "src/sealed_secret_ttrpc_async.rs")?;
+        fs::rename(
+            "src/sealed_secret_ttrpc.rs",
+            "src/sealed_secret_ttrpc_async.rs",
+        )?;
     }
 
     codegen(
make[1]: *** [Makefile:32: format] Error 1
make: *** [Makefile:33: check-libs] Error 2
```

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 1]
```diff
@@ -27,6 +27,8 @@ LIBC=${LIBC:-musl}
 # The kata agent enables seccomp feature.
 # However, it is not enforced by default: you need to enable that in the main configuration file.
 SECCOMP=${SECCOMP:-"yes"}
+# The kata agent enables sealed-secret feature.
+SEALED_SECRET=${SEALED_SECRET:-"yes"}
```
Comment (arronwy) [inline tools/osbuilder/rootfs-builder/rootfs.sh:31]: we may need set `SEALED_SECRET` default value to false, and enable it in `kata-deploy-binaries.sh` for `install_cc_image`

**Review by arronwy — APPROVED** [review]
Thanks @Lu-Biao  LGTM!

File: versions.yaml [hunk 2]
```diff
@@ -199,8 +199,8 @@ externals:
 
   attestation-agent:
     description: "Provide attested key unwrapping for image decryption"
-    url: "https://github.com/confidential-containers/guest-components/"
-    version: "v0.7.0"
+    url: "https://github.com/Xynnn007/guest-components.git"
+    version: "feat-cdh-binary"
```
Comment (arronwy) [inline versions.yaml:203]: will switch to upstream repo when https://github.com/confidential-containers/guest-components/pull/322 is merged

Comment (arronwy) [reply versions.yaml:203]: https://github.com/confidential-containers/guest-components/pull/322 is merged, please update to use the upstream tree

File: src/agent/src/main.rs [hunk 3]
```diff
@@ -354,6 +378,72 @@ async fn start_sandbox(
...
+        // launch CDH
+        let cdh_socket = Path::new(CDH_SOCKET);
+        if cdh_socket.exists() {
+            fs::remove_file(CDH_SOCKET)?;
+        }
+
+        Command::new(CDH_PATH).spawn()?;
+
+        // wait CDH boot
+        for _ in 0..20 {
```
Comment (fitzthum) [inline src/agent/src/main.rs:435]: How much time does this typically take?

File: src/agent/src/main.rs [hunk 4]
```diff
@@ -354,6 +378,72 @@ async fn start_sandbox(
...
+            if cdh_socket.exists() {
+                break;
+            }
+            info!(logger, "waiting for confidential-data-hub boot ...");
+            std::thread::sleep(std::time::Duration::from_secs(1));
+        }
```
Comment (fitzthum) [inline src/agent/src/main.rs:441]: Should we add some kind of catch here to log an error if we never find the socket?

File: src/agent/src/main.rs [hunk 5]
```diff
@@ -83,6 +85,24 @@ cfg_if! {
...
+const AA_UNIX_SOCKET_DIR: &str = "/run/confidential-containers/attestation-agent/";
+const UNIX_SOCKET_PREFIX: &str = "unix://";
+const AA_KEYPROVIDER_URI: &str =
+    concatcp!(UNIX_SOCKET_PREFIX, AA_UNIX_SOCKET_DIR, "keyprovider.sock");
+const AA_GETRESOURCE_URI: &str =
+    concatcp!(UNIX_SOCKET_PREFIX, AA_UNIX_SOCKET_DIR, "getresource.sock");
+const AA_ATTESTATION_SOCKET: &str = concatcp!(AA_UNIX_SOCKET_DIR, "attestation-agent.sock");
+const AA_ATTESTATION_URI: &str = concatcp!(UNIX_SOCKET_PREFIX, AA_ATTESTATION_SOCKET);
+
+cfg_if! {
```
Comment (fitzthum) [inline src/agent/src/main.rs:99]: I'm a bit confused by the relationship between the AA and the CDH here. I am guessing that this is just a temporary measure before we remove the keyprovider stuff from the AA? 

In the longterm won't the AA and CDH always be enabled together or are there certain platforms where we would only want one? Do we want to keep using the `aa-kbc-params` check to determine if they should both be enabled?

**Review by fitzthum — COMMENTED** [review]
Looks good. A few minor questions.

File: src/agent/src/main.rs (hunk 3 above)
Comment (Lu-Biao) [reply src/agent/src/main.rs]: typically  it's 1 sec.

File: src/agent/src/main.rs (hunk 4 above)
Comment (Lu-Biao) [reply src/agent/src/main.rs:436]: good suggestion,  I will update.

File: src/agent/src/main.rs (hunk 5 above)
Comment (Lu-Biao) [reply src/agent/src/main.rs:99]: yes, maybe after refactor of attestation-agent, CDH can call AA as a library, but current we need talk with AA directly.
we use aa_kbc_params as KBS config for both CDH and AA, but CDH depends on AA.

Comment (fitzthum) [reply src/agent/src/main.rs:99]: Ok. I guess this is fine for now.

**Comment by fidencio** [comment]
We're facing an issue with the tests right now, once those are fixed, this one is good to go, considering green tests.

**Comment by fidencio** [comment]
@Lu-Biao, I'm closing this one as all the commits are part of https://github.com/kata-containers/kata-containers/pull/7570 right now.

Thanks a lot, and please, let's follow up as part of https://github.com/kata-containers/kata-containers/pull/7570

**Pull request closed without merging**